### PR TITLE
DR-836 Remove check for presence of assets on dataset creation.

### PR DESCRIPTION
**Pull request by danxmoran**
Minimal changes needed to get assets out of the way.

### CODE REVIEW

File: src/test/java/bio/terra/app/controller/DatasetValidationsTest.java [hunk 1]
```diff
@@ -146,20 +146,6 @@ public void testDuplicateColumnNames() throws Exception {
...
-        req.getSchema().assets(null);
-        ErrorModel errorModel = expectBadDatasetCreateRequest(req);
-        checkValidationErrorModel("missingAssets", errorModel,
-            new String[]{"NotNull"});
-
-        req.getSchema().assets(Collections.emptyList());
-        errorModel = expectBadDatasetCreateRequest(req);
-        checkValidationErrorModel("missingAssets", errorModel,
-            new String[]{"NoAssets"});
-    }
```
Comment (danxmoran) [inline src/test/java/bio/terra/app/controller/DatasetValidationsTest.java:161]: I tried to replace this with a positive check, but started hitting errors about missing tokens etc. It looked like the backend might have actually been trying to create datasets when the request was valid. I looked into what it'd take to extract the validation logic into a more isolated space for unit testing without mocks, and it seemed like more work than I'd want to do as part of unblocking us here.

Comment (thathert) [reply src/test/java/bio/terra/app/controller/DatasetValidationsTest.java:161]: It is, we need a way to mock out the job service. I'll make a ticket, thanks for the fix!

**Review by thathert — APPROVED** [review]
LGTM

**Review by ddietterich — APPROVED** [review]
LGTM

**Review by marikomedlock — APPROVED** [review]
lgtm

**Pull request merged**